runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/jobsubmission/InMemoryJobServiceTest.java

```diff
@@ -186,7 +193,59 @@ public void testJobSubmissionUsesJobInvokerAndIsSuccess() throws Exception {
     verify(invocation, times(1)).start();
   }
 
+  @Test
+  public void testInvocationCleanup() {
+    final JobApi.GetJobsRequest getJobsRequest = JobApi.GetJobsRequest.newBuilder().build();
+
+    final int maxRunningJobs = maxInvocationHistory + new Random().nextInt(50);
```

Comment:
There are really only two cases to test, `maxRunningJobs <= maxInvocationHistory` and `maxRunningJobs > maxInvocationHistory`. The randomness is unnecessary IMO.

Reply:
I agree, not really necessary.